components/tools/OmeroPy/test/integration/test_ishare.py

```diff
@@ -983,6 +985,50 @@ def test8704(self):
         finally:
             user_client.__del__()
 
+    def test13018(self):
+        """
+        Test that image in share is unavailable when share
+        is inactive or expired
+        """
+        owner = self.new_client()
+        member, mobj = self.new_client_and_user()
+
+        createTestImage(owner.sf)
+        image = owner.sf.getQueryService().findAll("Image", None)[0]
+
+        o_share = owner.sf.getShareService()
+        sid = o_share.createShare("", None, [image], [mobj], [], True)
+
+        m_share = member.sf.getShareService()
+        m_share.activate(sid)
+
+        o_share.setActive(sid, False)
+        with pytest.raises(omero.ValidationException):
+            m_share.activate(sid)
+
+        with pytest.raises(omero.ValidationException):
+            obj = omero.model.ShareI(sid, False)
+            member.sf.setSecurityContext(obj)
+
+        # test inactive share, if member has no access to the image
+        s = o_share.getShare(sid)
+        with pytest.raises(Glacier2.PermissionDeniedException):
+            self.new_client(session=s.uuid)
+
+        # activate again
+        o_share.setActive(sid, True)
+
+        # test expired share, if member has no access to the image
+        expiration = long(time.time() * 1000) + 86400
```

Comment:
`expiration` is when the share *will* expire rather than it *has* expired. Server-side this is converted into the appropriate setting for `timeToLive`.